cmake/FindFreeRTOS.cmake

```diff
@@ -147,11 +153,34 @@ macro(stm32_find_freertos FreeRTOS_NAMESPACE FREERTOS_PATH)
             PATHS "${FreeRTOS_${PORT}_PATH}"
             NO_DEFAULT_PATH
         )
-        if(NOT (TARGET ${FreeRTOS_NAMESPACE}::${PORT}))
-            add_library(${FreeRTOS_NAMESPACE}::${PORT} INTERFACE IMPORTED)
-            target_link_libraries(${FreeRTOS_NAMESPACE}::${PORT} INTERFACE FreeRTOS)
-            target_sources(${FreeRTOS_NAMESPACE}::${PORT} INTERFACE "${FreeRTOS_${PORT}_SOURCE}")
-            target_include_directories(${FreeRTOS_NAMESPACE}::${PORT} INTERFACE "${FreeRTOS_${PORT}_PATH}")
+        if(NOT (TARGET ${FreeRTOS_NAMESPACE}::${PORT}${ARMv8_NON_SECURE}))
+            add_library(${FreeRTOS_NAMESPACE}::${PORT}${ARMv8_NON_SECURE} INTERFACE IMPORTED)
+            target_link_libraries(${FreeRTOS_NAMESPACE}::${PORT}${ARMv8_NON_SECURE} INTERFACE FreeRTOS)
+            target_sources(${FreeRTOS_NAMESPACE}::${PORT}${ARMv8_NON_SECURE} INTERFACE "${FreeRTOS_${PORT}_SOURCE}")
+            target_include_directories(${FreeRTOS_NAMESPACE}::${PORT}${ARMv8_NON_SECURE} INTERFACE "${FreeRTOS_${PORT}_PATH}")
+            message(trace "FindFreeRTOS: creating target ${FreeRTOS_NAMESPACE}::${PORT}${ARMv8_NON_SECURE}")
+            
+            # armv8-m needs additional file even if using "No Trust Zone" port
+            if(${PORT} STREQUAL ARM_CM23_NTZ OR ${PORT} STREQUAL ARM_CM33_NTZ OR ${PORT} STREQUAL ARM_CM23 OR ${PORT} STREQUAL ARM_CM33)
+                target_sources(${FreeRTOS_NAMESPACE}::${PORT}${ARMv8_NON_SECURE} INTERFACE "${FreeRTOS_${PORT}_PATH}/portasm.c")
+            endif()
+
+            if(${PORT} STREQUAL ARM_CM23 OR ${PORT} STREQUAL ARM_CM33)
+                # create the secure target
+                add_library(${FreeRTOS_NAMESPACE}::${PORT}::SECURE INTERFACE IMPORTED)
+                # ::SECURE doesn't link FreeRTOS like ::NON_SECURE does
+                target_sources(${FreeRTOS_NAMESPACE}::${PORT}::SECURE INTERFACE "${FreeRTOS_${PORT}_PATH}/../secure/secure_context.c"
+                                                                                "${FreeRTOS_${PORT}_PATH}/../secure/secure_context_port.c"
+                                                                                "${FreeRTOS_${PORT}_PATH}/../secure/secure_heap.c"
+                                                                                "${FreeRTOS_${PORT}_PATH}/../secure/secure_init.c")
+                message(trace "FindFreeRTOS: creating target ${FreeRTOS_NAMESPACE}::${PORT}::SECURE")
+
+                # non-secure part needs declaratation from secure includes
+                target_include_directories(${FreeRTOS_NAMESPACE}::${PORT}${ARMv8_NON_SECURE} INTERFACE "${FreeRTOS_${PORT}_PATH}/../secure")
```

Comment:
Since you need the `secure` folder so many times, It might be cleaner to find it, then avoid using the `../` part in all those paths.

Reply:
if portmacro.h has been found in /non_secure I think it's acce^table to not search for this path